core/shared/src/main/scala/com/lynbrookrobotics/potassium/streams/Stream.scala

```diff
@@ -494,6 +499,18 @@ abstract class Stream[+T] { self =>
     ret
   }
 
+  /**
+    * Returns a continuous even that is true when condition is true
+    * The pollingSource ImpulseEvent is used to check when condition is true
+    * @param condition condition of the event
+    * @return
+    */
+  def eventWithCondition(condition: T => Boolean): ContinuousEvent = {
+    val (event, updateEvent) = ContinuousEvent.newEvent
+    this.foreach(_ => updateEvent.apply(lastValue.exists(condition)))
```

Comment:
The cancellation function from listening needs to be stored somewhere inside the event needs to be stored somewhere so that the listening doesn't stop after GC.